### PR TITLE
Migrate org.elasticsearch.xpack to TransportVersion

### PR DESCRIPTION
**Pull request by thecoop**
Migrate most files in `org.elasticsearch.xpack` to use `TransportVersion`

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-2

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractor.java [hunk 1]
```diff
@@ -55,7 +54,7 @@ public CompositeKeyExtractor(String key, Property property, ZoneId zoneId, DataT
     CompositeKeyExtractor(StreamInput in) throws IOException {
         key = in.readString();
         property = in.readEnum(Property.class);
-        if (in.getVersion().onOrAfter(Version.fromId(INTRODUCING_UNSIGNED_LONG.id))) {
+        if (in.getTransportVersion().onOrAfter(INTRODUCING_UNSIGNED_LONG.transportVersion)) {
```
Comment (pgomulka) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractor.java:57]: how about migrating VersionCompatibilityChecks.INTRODUCING_UNSIGNED_LONG to TransportVersion?
the INTRODUCING_VERSION_FIELD_TYPE could be migrated too?

**Review by pgomulka — COMMENTED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractor.java (hunk 1 above)
Comment (thecoop) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractor.java:57]: Hmm, there's a more general thing about migrating `VersionCompatibilityChecks` class generally. I'll add it to the list

Comment (thecoop) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractor.java:57]: `VersionCompatibilityChecks` is used in `IndexCompatibility` as well - its not just for transport.

**Review by pgomulka — APPROVED** [review]
LGTM

**Pull request merged**